.github/workflows/poetry-version-check.yml

```diff
@@ -72,7 +72,8 @@ jobs:
           compare-to: ${{ needs.versions.outputs.version-base }}
           lenient: false
       - name: Head version newer than base
-        if: github.event_name != 'push' || github.ref != 'refs/heads/main' # don't run when pushing PR to main because then head==base
+        # Only compare versions for PRs, as that's the only time we have a different BASE and HEAD.
+        if: github.event_name == 'pull_request'
```

Comment:
Is the event_name not also pull_request if we merge a PR from a branch to main? In that case, the PR is merged FIRST, and then the actions run. But after the merge, BASE==HEAD already for the main branch.

Reply:
No I thought at that point the event name was `push` .

Here's the old constraint we had on the `publish` job:
```
      always()
      && needs.diff.outputs.src_changed > 0
      && !contains(needs.*.result, 'failure')
      && !contains(needs.*.result, 'cancelled')
      && (github.event_name == 'workflow_dispatch'
        || (github.event_name == 'push' && github.ref == 'refs/heads/main')
      )
```

The publish only ran when manually triggered with `workflow_dispatch` or when `github.event_name == push` for the main branch.

Reply:
Ah but I do think I need another `if: needs.diff.outputs.src_changed > 0` for the `version-checks`